web/src/components/RunDetailLayout/RunDetailLayout.tsx

```diff
@@ -42,6 +46,12 @@ const RunDetailLayout = ({test: {id, name, trigger}, test}: IProps) => {
   const [prevState, setPrevState] = useState(run.state);
   useDocumentTitle(`${name} - ${run.state}`);
 
+  const location = useLocation();
+  const [origin] = useState(() => {
+    const state = location.state as CustomizedState;
```

Comment:
Is there a need for the use state here?

Reply:
We need the state because we only want to extract the `origin` value the first time the page is loaded. Every time the `route` is updated the location object changes too.

Reply:
That's very interesting, what would be the advantages and disadvantages from `useMemo`?